pkg/fileutils/fileutils_test.go

```diff
@@ -335,7 +327,7 @@ func TestMatches(t *testing.T) {
 		{"dir/**", "dir/dir2/file", true},
 		{"dir/**", "dir/dir2/file/", true},
 		{"**/dir2/*", "dir/dir2/file", true},
-		{"**/dir2/*", "dir/dir2/file/", false},
+		{"**/dir2/*", "dir/dir2/file/", true},
```

Comment:
What 's kind of weird is that a `.dockerignore` with:
```
dir
!dir*/file
```
Where all I have is `dir/file` in my dir doesn't work. The above should "exclude dir" but then "include dir/file". But it doesn't. It excludes it all. If I change it to `!dir/file` file then it works.  This is true w/ and w/o your PR. 

Not directly related to this PR but in my testing of this I noticed it so I thought I'd mention it. I'll open an issue for it.